File: iosApp/iosApp/WebSocketWorker.swift
import SwiftUI
import UIKit
import ComposeApp

class WebSocketWorker: ObservableObject {
    private val webSocketService = IOSWebSocketService() // Твой WebSocketService
    private var backgroundTaskId: UIBackgroundTaskIdentifier = UIBackgroundTaskIdentifier.invalid

    init() {
        startBackgroundTask()
    }

    private func startBackgroundTask() {
        backgroundTaskId = UIApplication.shared.beginBackgroundTask {
            // Завершаем задачу, если система это требует
            UIApplication.shared.endBackgroundTask(self.backgroundTaskId)
            self.backgroundTaskId = UIBackgroundTaskIdentifier.invalid
        }

        // Логика для инициализации WebSocketService
        print("WebSocketWorker: WebSocketService initialized.")
    }

    func stopBackgroundTask() {
        if backgroundTaskId != UIBackgroundTaskIdentifier.invalid {
            UIApplication.shared.endBackgroundTask(backgroundTaskId)
            backgroundTaskId = UIBackgroundTaskIdentifier.invalid
        }
    }

    deinit {
        stopBackgroundTask()
    }

    // Добавим методы для взаимодействия с WebSocketService, если потребуется
    func getWebSocketService() -> WebSocketService {
        return webSocketService
    }
}

File: composeApp/src/iosMain/kotlin/com/jetbrains/kmpapp/feature/sockets/WebSocketWorker.kt
package com.jetbrains.kmpapp.feature.sockets

import platform.UIKit.UIApplication
import platform.UIKit.UIBackgroundTaskIdentifier
import platform.UIKit.UIBackgroundTaskInvalid
//import platform.UIKit
//import platform.UIKit.beginBackgroundTask
//import platform.UIKit.endBackgroundTask
import com.jetbrains.kmpapp.data.sockets.WebSocketService

class WebSocketWorker {
    private val webSocketService = IOSWebSocketService() // Реализация WebSocketService для iOS
    private var backgroundTaskId: UIBackgroundTaskIdentifier = UIBackgroundTaskInvalid

    init {
        startBackgroundTask()
    }

    private fun startBackgroundTask() {
        backgroundTaskId = UIApplication().beginBackgroundTaskWithName("WebSocketTask") {
            // Если задача завершилась, сообщаем системе, что работа окончена
            if (backgroundTaskId != UIBackgroundTaskInvalid) {
                UIApplication().endBackgroundTask(backgroundTaskId)
                backgroundTaskId = UIBackgroundTaskInvalid
            }
        }

        println("WebSocketWorker: WebSocketService initialized in background.")
    }

    fun stopBackgroundTask() {
        if (backgroundTaskId != UIBackgroundTaskInvalid) {
            UIApplication().endBackgroundTask(backgroundTaskId)
            backgroundTaskId = UIBackgroundTaskInvalid
        }
    }

    fun getWebSocketService(): WebSocketService = webSocketService
}
